### PR TITLE
Hasan/Pr 1319 smarttrader css setup

### PR DESCRIPTION
**Pull request by hasan-deriv**
* Setup tailwind css
* Setup shadcn/ui

### CODE REVIEW

File: src/main.tsx [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import React from "react";
 import ReactDOM from "react-dom/client";
 import App from "./App.tsx";
+import "./styles/main.css"
```
Comment (niloofar-deriv) [inline src/main.tsx]: Could u please add this folder in the configs (tsconfig + vite.config) so we can have absolute path

```suggestion
import "Styles/main.css"
```

Comment (hasan-deriv) [reply src/main.tsx]: sure

File: src/styles/fonts.css [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+/* ibm-plex-sans-500 - latin */
+@font-face {
+    font-display: swap; /* Check https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face/font-display for other options. */
+    font-family: 'IBM Plex Sans';
+    font-style: normal;
+    font-weight: 500;
+    src: url('./fonts/ibm-plex-sans/ibm-plex-sans-v19-latin-500.woff2') format('woff2'); /* Chrome 36+, Opera 23+, Firefox 39+, Safari 12+, iOS 10+ */
+}
```
Comment (niloofar-deriv) [inline src/styles/fonts.css]: Could u please add an empty line at the end of the file 🙏

File: postcss.config.cjs [hunk 3]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+    plugins: {
+        tailwindcss: {},
+        autoprefixer: {},
+    }
+}
```
Comment (niloofar-deriv) [inline postcss.config.cjs]: Could u please add an empty line at the end of the file 🙏

File: src/styles/main.css [hunk 4]
```diff
@@ -0,0 +1,24 @@
+@tailwind base;
+@tailwind components;
+@tailwind utilities;
+
+@layer base {
+    body {
+        @apply font-ibm text-prominent leading-body text-base font-normal overflow-x-hidden;
+    }
+    p {
```
Comment (niloofar-deriv) [inline src/styles/main.css:9]: Do we need this `mb`?

File: src/styles/main.css [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    h2,
+    h3,
+    h4,
+    h5,
+    h6{
+        @apply font-ibm text-prominent leading-base font-bold;
+    }
+    h3,
+    .h3 {
+        @apply text-[1.375rem];
```
Comment (niloofar-deriv) [inline src/styles/main.css:22]: And this one?

File: src/styles/main.css [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    }
+}
```
Comment (niloofar-deriv) [inline src/styles/main.css]: Could u please add an empty line at the end of the file 🙏

File: package.json [hunk 7]
```diff
@@ -11,8 +11,12 @@
     "test": "vitest"
   },
   "dependencies": {
```
Comment (niloofar-deriv) [inline package.json]: I guess you can add this too, https://tailwindcss.com/blog/automatic-class-sorting-with-prettier

File: components.json [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+  "tsx": true,
+  "tailwind": {
+    "config": "tailwind.config.js",
+    "css": "src/styles/main.css",
+    "baseColor": "slate",
+    "cssVariables": false
+  },
+  "aliases": {
+    "components": "Components/*",
+    "utils": "Utils/*"
```
Comment (niloofar-deriv) [inline components.json:14]: Do we need to define the rest of the folders here?

File: components.json [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+  }
+}
```
Comment (niloofar-deriv) [inline components.json]: Could u please add an empty line here 🙏

File: tailwind.config.js [hunk 10]
```diff
@@ -0,0 +1,88 @@
+/** @type {import('tailwindcss').Config} */
+module.exports = {
```
Comment (niloofar-deriv) [inline tailwind.config.js:2]: Are these styles base on DSmartTrader?

**Review by mayuran-deriv — APPROVED** [review]

File: components.json (hunk 8 above)
Comment (hasan-deriv) [reply components.json:14]: I don't think so. shadcn only need these two

File: components.json (hunk 9 above)
Comment (hasan-deriv) [reply components.json]: Sure

File: postcss.config.cjs (hunk 3 above)
Comment (hasan-deriv) [reply postcss.config.cjs]: Sure

File: src/styles/fonts.css (hunk 2 above)
Comment (hasan-deriv) [reply src/styles/fonts.css]: sure

File: src/styles/main.css (hunk 4 above)
Comment (hasan-deriv) [reply src/styles/main.css:9]: Yeah, i saw in production ever `p` has margin bottom.

File: src/styles/main.css (hunk 5 above)
Comment (hasan-deriv) [reply src/styles/main.css:22]: all `h3` has this font size.

File: src/styles/main.css (hunk 6 above)
Comment (hasan-deriv) [reply src/styles/main.css]: Sure..

File: tailwind.config.js (hunk 10 above)
Comment (hasan-deriv) [reply tailwind.config.js:2]: Yeah

File: package.json (hunk 7 above)
Comment (hasan-deriv) [reply package.json]: Sure

**Review by niloofar-deriv — APPROVED** [review]

File: package.json [hunk 11]
```diff
@@ -11,8 +11,12 @@
...
+    "class-variance-authority": "^0.7.0",
+    "clsx": "^2.0.0",
```
Comment (matin-deriv) [inline package.json]: Can you elaborate more on the usage of clsx? I'm wondering if we can make use of Tailwind classes to handle conditional classnames!

File: tailwind.config.js [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+              maxWidth: "960px",
+              marginInline: "auto",
+              paddingInline: "10px",
+              position: "relative",
+              zIndex: 10,
+          },
+      });
+    },
+  ],
+}
```
Comment (matin-deriv) [inline tailwind.config.js]: Can you add the last empty line? you can checkout your prettier setup 🙏🏼

File: src/styles/fonts.css (hunk 2 above)
Comment (nada-deriv) [reply src/styles/fonts.css:53]: can add css extension as well in the prettier script. as of now, i had included only ts, tsx, js, and jsx.

**Review by nada-deriv — APPROVED** [review]

**Review by niloofar-deriv — APPROVED** [review]

File: package.json (hunk 11 above)
Comment (hasan-deriv) [reply package.json]: We can make conditional classNames more readable and easier with it.

```
        <div
            className={clsx(
                "alert tw-py-2.5 tw-pr-3 tw-pl-3 tw-rounded nextIcon:tw-mr-2",
                color === "light" && "tw-bg-gray-200 nextIcon:tw-text-azure",
                color === "warning" && "tw-bg-warning-100 tw-text-heading",
                color === "secondary" && "tw-bg-secondary tw-text-white",
                className
            )}
        >
            {children}
        </div>
```

**Review by niloofar-deriv — APPROVED** [review]
LGTM 👍

**Review by mayuran-deriv — APPROVED** [review]

**Review by matin-deriv — APPROVED** [review]

**Pull request merged**